AIDevGallery/Utils/OllamaHelper.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace AIDevGallery.Utils;
+
+internal record OllamaModel(string Name, string Tag, string Id, string Size, string Modified);
+
+internal class OllamaHelper
+{
+    private static bool? isOllamaAvailable;
+    public static List<OllamaModel>? GetOllamaModels()
+    {
+        if (isOllamaAvailable != null && !isOllamaAvailable.Value)
+        {
+            return null;
+        }
+
+        try
+        {
+            using (var p = new Process())
+            {
+                p.StartInfo.FileName = "ollama";
+                p.StartInfo.Arguments = "list";
+                p.StartInfo.RedirectStandardOutput = true;
+                p.StartInfo.RedirectStandardError = true;
+                p.StartInfo.UseShellExecute = false;
+                p.StartInfo.CreateNoWindow = true;
+
+                p.Start();
+                p.WaitForExit();
+
+                var lines = new List<string>();
+
+                Task.WhenAny(
```

Comment:
Why is all of this needed? If the process has closed already, can't you just read the output?